### PR TITLE
feat(account-lib): add ada transaction builder

### PR DESCRIPTION
**Pull request by ewangbitgo**
Ticket: BG-50796

### CODE REVIEW

**Review by margueriteblair — APPROVED** [review]
lgtm

File: modules/sdk-coin-ada/src/transaction.ts [hunk 1]
```diff
@@ -1,17 +1,185 @@
-import { BaseTransaction, BaseKey, NotImplementedError } from '@bitgo/sdk-core';
+import {
+  BaseKey,
+  BaseTransaction,
+  Entry,
+  InvalidTransactionError,
+  TransactionType,
+  TransactionExplanation,
+} from '@bitgo/sdk-core';
+import * as CardanoWasm from '@emurgo/cardano-serialization-lib-nodejs';
```
Comment (byhow) [inline modules/sdk-coin-ada/src/transaction.ts:9]: How will we be handling using the sdk in the browser given that this seems to be a nodejs package?

File: modules/sdk-coin-ada/src/transactionBuilderFactory.ts [hunk 2]
```diff
@@ -11,8 +11,21 @@ export class TransactionBuilderFactory extends BaseTransactionBuilderFactory {
   }
 
   /** @inheritdoc */
-  from(raw: Uint8Array | string): TransactionBuilder {
-    throw new NotImplementedError('from not implemented');
+  from(raw: string): TransactionBuilder {
```
Comment (byhow) [inline modules/sdk-coin-ada/src/transactionBuilderFactory.ts:14]: Consider adding BREAKING CHANGE or exclamation marks after commit type as this changes how function parameters are passed in

File: modules/sdk-coin-ada/test/unit/utils.ts [hunk 3]
```diff
@@ -27,6 +27,8 @@ describe('utils', () => {
   });
 
   it('should validate public key correctly', () => {
+    //    should.equal(utils.isValidPublicKey('0304e96b46fcb0974e94a002fda63f3ba8b97429ed27238b7b7ad52553219d4928'), true);
```
Comment (byhow) [inline modules/sdk-coin-ada/test/unit/utils.ts:30]: nit: remove comments if not needed

File: modules/sdk-coin-ada/src/transactionBuilderFactory.ts (hunk 2 above)
Comment (mmcshinsky-bitgo) [reply modules/sdk-coin-ada/src/transactionBuilderFactory.ts:14]: Side note, BitGoJS fails the `!`, so we have to use BREAKING CHANGE in the commit footer.

File: modules/sdk-coin-ada/src/transaction.ts (hunk 1 above)
Comment (margueriteblair) [reply modules/sdk-coin-ada/src/transaction.ts:9]: We wrap it within @cardano-serialization-lib-browser which is compatible :) The polyfill code for that is in bitgo.config.js

File: modules/sdk-coin-ada/src/transaction.ts [hunk 4]
```diff
@@ -1,17 +1,185 @@
-import { BaseTransaction, BaseKey, NotImplementedError } from '@bitgo/sdk-core';
+import {
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/src/transaction.ts:1]: Let's move all transaction/transfer files to `lib` for consistency with other coins.

File: modules/sdk-coin-ada/src/transactionBuilderFactory.ts (hunk 2 above)
Comment (byhow) [reply modules/sdk-coin-ada/src/transactionBuilderFactory.ts:14]: Ah good to know

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**